core/src/main/java/org/eclipse/hono/config/AbstractConfig.java

```diff
@@ -41,25 +41,23 @@
     /**
      *  A logger to be shared with subclasses.
      */
-    protected final Logger LOG = LoggerFactory.getLogger(getClass());
+    public static final Logger LOG = LoggerFactory.getLogger(AbstractConfig.class);
 
-    private String trustStorePath;
-    private char[] trustStorePassword;
-    private TrustOptions trustOptions;
+    private String trustStorePath = "";
+    private String trustStorePassword = "";
```

Comment:
I think it's common practice to rather use `char[]` for password fields.

Reply:
As all other libraries we are passing these passwords to are keeping them as Strings in memory, I don't think we're getting any benefits from keeping them as char arrays. 
And if we still chose to do so, I think getter/setter methods should follow that as well. 
Anyway, in my opinion we're just keeping a lot of transformation code that gives us no real benefits.

Reply:
Ok.